apps/web/modules/videos/views/videos-single-view.tsx

```diff
@@ -23,111 +22,98 @@ import { CalAiTranscribe } from "~/videos/ai/ai-transcribe";
 export type PageProps = inferSSRProps<typeof getServerSideProps>;
 
 export default function JoinCall(props: PageProps) {
-  const { t } = useLocale();
   const { meetingUrl, meetingPassword, booking, hasTeamPlan, calVideoLogo } = props;
   const [daily, setDaily] = useState<DailyCall | null>(null);
 
   useEffect(() => {
-    const callFrame = DailyIframe.createFrame({
-      theme: {
-        colors: {
-          accent: "#FFF",
-          accentText: "#111111",
-          background: "#111111",
-          backgroundAccent: "#111111",
-          baseText: "#FFF",
-          border: "#292929",
-          mainAreaBg: "#111111",
-          mainAreaBgAccent: "#1A1A1A",
-          mainAreaText: "#FFF",
-          supportiveText: "#FFF",
-        },
-      },
-      showLeaveButton: true,
-      iframeStyle: {
-        position: "fixed",
-        width: "100%",
-        height: "100%",
-      },
-      url: meetingUrl,
-      ...(typeof meetingPassword === "string" && { token: meetingPassword }),
-      ...(hasTeamPlan && {
-        customTrayButtons: {
-          recording: {
-            label: "Record",
-            tooltip: "Start or stop recording",
-            iconPath: RECORDING_DEFAULT_ICON,
-            iconPathDarkMode: RECORDING_DEFAULT_ICON,
-          },
-          transcription: {
-            label: "Cal.ai",
-            tooltip: "Transcription powered by AI",
-            iconPath: TRANSCRIPTION_STOPPED_ICON,
-            iconPathDarkMode: TRANSCRIPTION_STOPPED_ICON,
+    let callFrame: DailyCall | undefined;
+    try {
+      callFrame = DailyIframe.createFrame({
+        theme: {
+          colors: {
+            accent: "#FFF",
+            accentText: "#111111",
+            background: "#111111",
+            backgroundAccent: "#111111",
+            baseText: "#FFF",
+            border: "#292929",
+            mainAreaBg: "#111111",
+            mainAreaBgAccent: "#1A1A1A",
+            mainAreaText: "#FFF",
+            supportiveText: "#FFF",
           },
         },
-      }),
-    });
-
-    setDaily(callFrame);
+        showLeaveButton: true,
+        iframeStyle: {
+          position: "fixed",
+          width: "100%",
+          height: "100%",
+        },
+        url: meetingUrl,
+        ...(typeof meetingPassword === "string" && { token: meetingPassword }),
+        ...(hasTeamPlan && {
+          customTrayButtons: {
+            recording: {
+              label: "Record",
+              tooltip: "Start or stop recording",
+              iconPath: RECORDING_DEFAULT_ICON,
+              iconPathDarkMode: RECORDING_DEFAULT_ICON,
+            },
+            transcription: {
+              label: "Cal.ai",
+              tooltip: "Transcription powered by AI",
+              iconPath: TRANSCRIPTION_STOPPED_ICON,
+              iconPathDarkMode: TRANSCRIPTION_STOPPED_ICON,
+            },
+          },
+        }),
+      });
+    } catch (err) {
+      callFrame = DailyIframe.getCallInstance();
```

Comment:
This refactor is needed to avoid "Duplicate DailyIframe instances are not allowed" error.

Seems like a common case. cc https://www.npmjs.com/package/@daily-co/daily-js/v/0.44.0
<img width="770" alt="Screenshot 2024-12-11 at 12 07 49 AM" src="https://github.com/user-attachments/assets/026fa244-61f9-4e68-8f4d-6caf7b5cc9d9">